### ISSUE TITLE
Conflicting property name definitions

### ISSUE THREAD

**Issue by sgjohnston** [FasterXML/jackson-databind#193]
I have an awkward problem caused by obfuscation of a Java class, when I then try to serialize it with Jackson. The properties are all annotated, so the problem isn't the obvious obfuscation one. In fact, I can recreate it without obfuscation, with a pretty artificial example. The basic issue is is if you have a field and a getter method which both have the same name, and they have a @JsonProperty annotation to indicate that they are different Json properties, then serialization will fail with an exception, indicating that there is a conflicting property name definition.

I could understand the reverse - a field and method with the same Json property name, when the serializer or deserializer cannot decide which to use, but in the direction I have the problem, I can't see what the problem is. I have looked at the jackson code, but I can see that trying to figure this out would be a long task. Possibly a bug/limitation? Btw, I can't assume that a field and method won't have the same name because of the obfuscation used.

Here is the sample which recreates:

```
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.ObjectMapper;

public class Test
{

  private static final String jsonValue1 = "val1";
  private static final String jsonValue2 = "val2";

  @JsonProperty(jsonValue1)
  private int x;
  @JsonIgnore
  private int value2;

  public Test(@JsonProperty(jsonValue1)int value1,
            @JsonProperty(jsonValue2)int value2)
{
    this.x = value1;
    this.value2 = value2;
}

@JsonProperty(jsonValue2)
int x()
{
    return value2;
}

  public String toString() {
    return "value = "+x+", value 2 = "+value2;
  }

  public static void main(String[] args)
  {
    try {
        Test t = new Test(123, 456);
        ObjectMapper mapper = new ObjectMapper();
        String serialised = mapper.writeValueAsString(t);
        System.out.println("Serialised: "+serialised);
    }
    catch (Exception ex) {
        System.out.println("!!Exception: "+ex);
        ex.printStackTrace();
        System.out.println("Cause: ");
        ex.getCause().printStackTrace();
    }
  }
}
```

and the thrown exception:

```
!!Exception: java.lang.IllegalStateException: Conflicting property name definitions: 'val1' (for [field Test#x]) vs 'val2' (for [method x(0 params)])
```

java.lang.IllegalStateException: Conflicting property name definitions: 'val1' (for [field Test#x]) vs 'val2' (for [method Test#x(0 params)])
Cause: 
at com.fasterxml.jackson.databind.introspect.POJOPropertyBuilder.findRenamed(POJOPropertyBuilder.java:580)
at com.fasterxml.jackson.databind.introspect.POJOPropertyBuilder.findNewName(POJOPropertyBuilder.java:557)
at com.fasterxml.jackson.databind.introspect.POJOPropertiesCollector._renameProperties(POJOPropertiesCollector.java:648)
at com.fasterxml.jackson.databind.introspect.POJOPropertiesCollector.collect(POJOPropertiesCollector.java:240)
at com.fasterxml.jackson.databind.introspect.BasicClassIntrospector.collectProperties(BasicClassIntrospector.java:142)
at com.fasterxml.jackson.databind.introspect.BasicClassIntrospector.forSerialization(BasicClassIntrospector.java:68)
at com.fasterxml.jackson.databind.introspect.BasicClassIntrospector.forSerialization(BasicClassIntrospector.java:11)
at com.fasterxml.jackson.databind.SerializationConfig.introspect(SerializationConfig.java:490)
at com.fasterxml.jackson.databind.ser.BeanSerializerFactory.createSerializer(BeanSerializerFactory.java:131)
at com.fasterxml.jackson.databind.ser.SerializerFactory.createSerializer(SerializerFactory.java:53)
at com.fasterxml.jackson.databind.SerializerProvider._createUntypedSerializer(SerializerProvider.java:935)
at com.fasterxml.jackson.databind.SerializerProvider._createAndCacheUntypedSerializer(SerializerProvider.java:892)
at com.fasterxml.jackson.databind.SerializerProvider.findValueSerializer(SerializerProvider.java:429)
at com.fasterxml.jackson.databind.SerializerProvider.findTypedValueSerializer(SerializerProvider.java:520)
at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider.serializeValue(DefaultSerializerProvider.java:99)
at com.fasterxml.jackson.databind.ObjectMapper._configAndWriteValue(ObjectMapper.java:2626)
at com.fasterxml.jackson.databind.ObjectMapper.writeValueAsString(ObjectMapper.java:2118)
at Test.main(Test.java:54)

**Comment by unexpected62** [comment]
@cowtowncoder - is https://github.com/nestlabs/android-sdk/issues/16 the same problem here? That project is using jackson-2.6.4 from what I can tell: https://github.com/nestlabs/android-sdk/blob/4e0f888b12db3e34c6c0bc91374e1ca0037711c5/sdk/build.gradle#L198-L199

**Comment by cowtowncoder** [comment]
@unexpected62 I have no idea; it is typically difficult to comment on old issues that were fixed. If you could file a new issue with information you included it would be easier to track. Versions before 2.6 are not actively maintained any more, so reproduction with 2.7(.7) would be ideal -- no fixes will be backported for 2.6.

**Comment by unexpected62** [comment]
@cowtowncoder - OK, thanks for the quick update. I should see if I can get Nest to upgrade to Jackson 2.7+
